server/src/test/java/org/elasticsearch/snapshots/SnapshotsServiceTests.java

```diff
@@ -505,16 +505,22 @@ private static ClusterState stateWithSnapshots(String repository, SnapshotsInPro
         );
     }
 
-    private static void assertIsNoop(ClusterState state, SnapshotsService.ShardSnapshotUpdate shardCompletion) throws Exception {
+    private static void assertIsNoop(ClusterState state, SnapshotsService.SnapshotTask shardCompletion) throws Exception {
         assertSame(applyUpdates(state, shardCompletion), state);
     }
 
-    private static ClusterState applyUpdates(ClusterState state, SnapshotsService.ShardSnapshotUpdate... updates) throws Exception {
-        return ClusterStateTaskExecutorUtils.executeAndAssertSuccessful(
-            state,
-            SnapshotsService.SHARD_STATE_EXECUTOR,
-            Arrays.asList(updates)
-        );
+    private static ClusterState applyUpdates(ClusterState state, SnapshotsService.SnapshotTask... updates) throws Exception {
```

Comment:
Could we use the real executor here? I think I'd prefer that.

Reply:
Not easily it turns out, need a snapshot service instance for that and we don't have one in these tests (weirdly enough despite the tests name) and it's quite bothersome to add one it turns out, we'd need a test-only field just for the executor or something along those lines.
I know it's crappy but these tests don't seem to be worth it going the extra mile here to make things nice code when code might change again soon?